docs/examples/checkpointing/job.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+#SBATCH --gpus-per-task=1
+#SBATCH --cpus-per-task=4
+#SBATCH --ntasks-per-node=1
+#SBATCH --mem=16G
+#SBATCH --time=00:15:00
+#SBATCH --requeue
+#SBATCH --signal=B:TERM@300 # tells the controller to send SIGTERM to the job 5
+                            # min before its time ends to give it a chance for
+                            # better cleanup. If you cancel the job manually,
+                            # make sure that you specify the signal as TERM like
+                            # so `scancel --signal=TERM <jobid>`.
+                            # https://dhruveshp.com/blog/2021/signal-propagation-on-slurm/
+
+# trap the signal to the main BATCH script here.
+sig_handler()
+{
+	echo "BATCH interrupted"
+	wait # wait for all children, this is important!
+}
+
+trap 'sig_handler' SIGINT SIGTERM SIGCONT
```

Comment:
Useless after `exec python main.py`

Reply:
Yeah you're right! Thanks @obilaniu. Should we be encouraging users to do `srun python (...)` or `exec python (...)` in your view?

Reply:
This being an example only, I would adopt one of two strategies - either minimize the delta from the foundation mini-example to this one, or minimize the complexity of this one in the absolute.

I would prefer that you drop the `trap` business, and use the Python equivalent (the `signal` package).

However, I have no faith in asynchronous checkpointing at all and recommend only synchronous checkpointing. I would use the `signal` package only to demonstrate in the mini-example that Python can install signal handlers and execute Python code as a result.

Reply:
Ok, here's my take: **779413b8fc32907607f268c55391dc839965a75a**